### PR TITLE
[ENG-644] [ENG-645] - Add onboarding checklist

### PR DESCRIPTION
**Pull request by floreks**
## Summary

**IMPORTANT**: All existing users are marked by the API backfill logic as users that finished the checklist. It will only appear for new users after finishing the onboarding (or skipping the demo).

Linear: [ENG-644](https://linear.app/pluralsh/issue/ENG-644/implement-checklist-to-apppluralsh-onboarding-flow) | [ENG-645](https://linear.app/pluralsh/issue/ENG-645/add-notification-functionality-for-dismissed-checklist)

image

image

### Tested scenarios
Besides the below tests, I have also checked the "re-appear one time after dismiss". It works in a way that after initial dismiss, the user has to log out and back in to see the checklist again. After that dismiss will be permanent.

#### BYOC - Local CLI
1. Followed the onboarding and selected local CLI setup
1. Executed `plural init`
1. First checklist step was automatically marked as done 
1. Installed few applications
1. Checklist was completed and completed step was shown

#### BYOC - Cloud Shell
1. Followed the onboarding and select cloud shell setup
1. After opening cloud shell
   1. skipped demo - first step was automatically marked as done
   2. installed airbyte and console using quickstack - checklist was automatically completed

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: www/src/components/audits/Audits.js
Comment (floreks) [inline www/src/components/audits/Audits.js:47]: This is a small fix since we do not have `repositories` route registered and the audit links weren't working.

File: www/src/components/layout/WithNotifications.tsx
Comment (maciaszczykm) [inline www/src/components/layout/WithNotifications.tsx:73]: nit: Formatting is a bit off.

File: www/src/components/shell/onboarding/checklist/Footer.tsx
Comment (maciaszczykm) [inline www/src/components/shell/onboarding/checklist/Footer.tsx:63]: `setDismissed` and `setDismiss` names are a bit confusing.

File: www/src/components/layout/WithNotifications.tsx
Comment (maciaszczykm) [inline www/src/components/layout/WithNotifications.tsx:51]: nit: It would be good to break into more lines.

File: www/src/components/layout/WithNotifications.tsx
Comment (maciaszczykm) [inline www/src/components/layout/WithNotifications.tsx:50]: Really like usage of generated types!

**Review by maciaszczykm — COMMENTED** [review]
Code looks good, good job. I've left some minor comments.

**Comment by maciaszczykm** [comment]
I'd wait for one more approval before merging as this is quite important feature.

**Comment by maciaszczykm** [comment]
One more thing, button alt text seems to be invalid here:
Zrzut ekranu 2022-10-21 o 13 58 12

**Comment by maciaszczykm** [comment]
I've also noticed that when checklist appeared also navigation bar on the left appeared. Is this expected?

Zrzut ekranu 2022-10-21 o 13 57 25

**Comment by maciaszczykm** [comment]
Also after testing it with CLI (PROD-16) I can confirm that installing console automatically checked second step, same with third step and other app (Grafana OnCall in my case). For the last step I had to refresh the page:

Zrzut ekranu 2022-10-21 o 16 04 35

**Comment by floreks** [comment]
> Also after testing it with CLI (PROD-16) I can confirm that installing console automatically checked second step, same with third step and other app (Grafana OnCall in my case). For the last step I had to refresh the page:
> 
> Zrzut ekranu 2022-10-21 o 16 04 35

Oh yeah, you'd need to wait a bit. First API has to trigger the transition and then the frontend has to trigger the refresh (it's done every 60s). I would be updated after some time.

**Comment by floreks** [comment]
> I've also noticed that when checklist appeared also navigation bar on the left appeared. Is this expected?
> 
> Zrzut ekranu 2022-10-21 o 13 57 25
> 
> Edit note: I was still during onboarding flow when this happened.

Not sure if this is related to the checklist itself as I haven't touched the sidebar. Maybe user is marked as onboarded automatically when doing some CLI actions?

File: www/src/components/shell/onboarding/checklist/Footer.tsx
Comment (michaeljguarino) [inline www/src/components/shell/onboarding/checklist/Footer.tsx:73]: we should add these variables to the hook right?

File: www/src/components/layout/WithNotifications.tsx
Comment (michaeljguarino) [inline www/src/components/layout/WithNotifications.tsx:123]: Also a case where mutation attributes are more readable when added to the hook, I think

Comment (floreks) [reply www/src/components/layout/WithNotifications.tsx:123]: oh yea, previously there were actually 2 separate mutation calls with different values, that's why it's here.

**Review by zreigz — APPROVED** [review]

**Pull request merged**
